addon/scheduler.ts

```diff
@@ -60,21 +63,23 @@ export function endTransition(): void {
  * Correct connects the router's transition events to
  * app scheduler's work.
  *
- * @param {Router} router An instance of an Ember router.
+ * @param {RouterService|Router} router An instance of a RouterService or an Ember Router.
  */
-export function setupRouter(router: Router): void {
+export function setupRouter(router: RouterService | Router): void {
```

Comment:
What do you think about adding a deprecation here if users are using the `Router`? This would indicate that they're doing the manual registration vs. the router service. We can add this deprecation now, and schedule it until the following major (following this release).

Reply:
I've added a deprecation warning for the use of `Router`.